shared/actions/devices.js

```diff
@@ -82,33 +83,54 @@ export function generatePaperKey () : AsyncAction {
   }
 }
 
-export function removeDevice (deviceID: string) : AsyncAction {
-  return navigateUpOnUnchanged((dispatch, getState, maybeNavigateUp) => {
-    const params : revoke_revokeDevice_rpc = {
-      method: 'revoke.revokeDevice',
-      param: {deviceID, force: false},
-      incomingCallMap: {},
-      callback: error => {
-        dispatch({
-          type: Constants.deviceRemoved,
-          payload: error,
-          error: !!error
-        })
-
-        // TODO when device page is integrated
-        // if (wasCurrentDevice) {
-          // dispatch(setRevokedSelf(oldCurrentDeviceName))
-          // dispatch(navigateTo('', loginTab))
-          // dispatch(switchTab(loginTab))
-        // } else
-
-        if (!error) {
-          dispatch(loadDevices())
-          maybeNavigateUp()
+export function removeDevice (deviceID: string, name: string, currentDevice: boolean): AsyncAction {
```

Comment:
Have to use a different 'deprovision' RPC for the special case of revoking the current device.